experiments/AMIP/components/atmosphere/climaatmos.jl

```diff
@@ -100,48 +115,68 @@ function Interfacer.get_field(atmos_sim::ClimaAtmosSimulation, ::Val{:radiative_
 
         return @. -(LWd_TOA + SWd_TOA - LWu_TOA - SWu_TOA)
     else
-        return FT(0)
+        return zero_scalar_half(atmos_sim.integrator)
     end
 end
 
 function Interfacer.get_field(atmos_sim::ClimaAtmosSimulation, ::Val{:energy})
-    thermo_params = get_thermo_params(atmos_sim)
+    integrator = atmos_sim.integrator
+    p = integrator.p
 
-    ᶜS_ρq_tot = atmos_sim.integrator.p.precipitation.ᶜS_ρq_tot
-    ᶜts = atmos_sim.integrator.p.precomputed.ᶜts
-    ᶜΦ = atmos_sim.integrator.p.core.ᶜΦ
 
     # return total energy and (if Microphysics0Moment) the energy lost due to precipitation removal
-    if atmos_sim.integrator.p.atmos.precip_model isa CA.Microphysics0Moment
-        ᶜS_ρq_tot = atmos_sim.integrator.p.precipitation.ᶜS_ρq_tot
-        ᶜts = atmos_sim.integrator.p.precomputed.ᶜts
-        ᶜΦ = atmos_sim.integrator.p.core.ᶜΦ
-        return atmos_sim.integrator.u.c.ρe_tot .-
-               ᶜS_ρq_tot .* CA.e_tot_0M_precipitation_sources_helper.(Ref(thermo_params), ᶜts, ᶜΦ) .*
-               atmos_sim.integrator.dt
+    if p.atmos.precip_model isa CA.Microphysics0Moment
+        ᶜts = p.precomputed.ᶜts
+        ᶜΦ = p.core.ᶜΦ
+        ᶜS_ρq_tot = p.precipitation.ᶜS_ρq_tot
+        thermo_params = get_thermo_params(atmos_sim)
+        return integrator.u.c.ρe_tot .-
+               ᶜS_ρq_tot .* CA.e_tot_0M_precipitation_sources_helper.(Ref(thermo_params), ᶜts, ᶜΦ) .* integrator.dt
     else
-        return atmos_sim.integrator.u.c.ρe_tot
+        return integrator.u.c.ρe_tot
     end
 end
 
+# helpers for get_field extensions
```

Comment:
It might be helpful to add a comment that we have two cases for moisture (dry vs moist) and two for radiation (without or with), which is why we have two versions of each function here